### PR TITLE
[vsts-938452] Add `altool` upload and verify draft implementations

### PR DESCRIPTION
**Pull request by olegoid**
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/938452

### CODE REVIEW

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        public string Password { get ;set; }
+
+        [Required]
+        public string FilePath { get ;set; }
+
+        [Required]
+        public PlatformName FileType { get ; set; }
+
+        protected override string ToolName {
+			get { return "altool"; }
```
Comment (rolfbjarne) [inline msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs]: The indentation is wrong in several places in this file.

File: msbuild/Xamarin.MacDev.Tasks.Core/Xamarin.MacDev.Tasks.Core.csproj [hunk 2]
```diff
@@ -1,4 +1,3 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (rolfbjarne) [inline msbuild/Xamarin.MacDev.Tasks.Core/Xamarin.MacDev.Tasks.Core.csproj]: Unnecessary whitespace change.

File: msbuild/Xamarin.MacDev.Tasks.Core/Xamarin.MacDev.Tasks.Core.csproj [hunk 3]
```diff
@@ -118,6 +117,7 @@
     <Compile Include="ProcessUtils.cs" />
     <Compile Include="StringParserService.cs" />
     <Compile Include="../../tools/common/FileCopier.cs" />
+    <Compile Include="Tasks\AlToolTaskBase.cs" />
   </ItemGroup>
   <Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />
-</Project>
+</Project>
```
Comment (rolfbjarne) [inline msbuild/Xamarin.MacDev.Tasks.Core/Xamarin.MacDev.Tasks.Core.csproj]: Unnecessary whitespace change.

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+			return args.ToString ();
+		}
+
+        string GetFileTypeValue ()
+        {
+            switch(FileType) {
+                case PlatformName.MacOSX: return "osx";
+                case PlatformName.TvOS: return "appletvos";
+                case PlatformName.iOS: return "ios";
+                default: throw new NotSupportedException ("Provided file type is not supported by altool");
```
Comment (rolfbjarne) [inline msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs]:
```suggestion
                default: throw new NotSupportedException ($"Provided file type '{FileType}' is not supported by altool");
```

File: msbuild/Xamarin.Mac.Tasks.Core/Tasks/ALToolUploadTaskBase.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
+using System.Text;
+
+using Xamarin.MacDev;
+
+namespace Xamarin.Mac.Tasks
+{
+	public class ALToolUploadTaskBase : Xamarin.MacDev.Tasks.ALToolTaskBase
+	{
+        public override PlatformName FileType => PlatformName.MacOSX;
```
Comment (emaf) [inline msbuild/Xamarin.Mac.Tasks.Core/Tasks/ALToolUploadTaskBase.cs]: Wrong indentation

File: msbuild/Xamarin.Mac.Tasks.Core/Tasks/ALToolValidateTaskBase.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
+using System.Text;
+
+using Xamarin.MacDev;
+
+namespace Xamarin.Mac.Tasks
+{
+	public class ALToolValidateTaskBase : Xamarin.MacDev.Tasks.ALToolTaskBase
+	{
+        public override PlatformName FileType => PlatformName.MacOSX;
```
Comment (emaf) [inline msbuild/Xamarin.Mac.Tasks.Core/Tasks/ALToolValidateTaskBase.cs]: Same indentation problem here

File: msbuild/Xamarin.iOS.Tasks.Core/Tasks/ALToolUploadTaskBase.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
+using System.Text;
+
+using Xamarin.MacDev;
+
+namespace Xamarin.iOS.Tasks
+{
+	public class ALToolUploadTaskBase : Xamarin.MacDev.Tasks.ALToolTaskBase
+	{
+        public override PlatformName FileType => PlatformName.iOS;
```
Comment (emaf) [inline msbuild/Xamarin.iOS.Tasks.Core/Tasks/ALToolUploadTaskBase.cs]: Same problem here

File: msbuild/Xamarin.iOS.Tasks.Core/Tasks/ALToolVerifyTaskBase.cs [hunk 8]
```diff
@@ -0,0 +1,20 @@
+using System.Text;
+
+using Xamarin.MacDev;
+
+namespace Xamarin.iOS.Tasks
+{
+	public class ALToolValidateTaskBase : Xamarin.MacDev.Tasks.ALToolTaskBase
+	{
+        public override PlatformName FileType => PlatformName.iOS;
```
Comment (emaf) [inline msbuild/Xamarin.iOS.Tasks.Core/Tasks/ALToolVerifyTaskBase.cs]: Same problem here
